test/jb_funding_cycle_store/configure_for.test.js

```diff
@@ -0,0 +1,1396 @@
+    const firstFundingCycleMetadata = ethers.BigNumber.from(123);
+
+    // Configure first funding cycle
+    const firstConfigureForTx = await jbFundingCycleStore
+      .connect(controller)
+      .configureFor(PROJECT_ID, firstFundingCycleData, firstFundingCycleMetadata, fundingCycleMustStartOnOrAfterZero);
+
+    // The timestamp the first configuration was made during.
+    const firstConfigurationTimestamp = await getTimestamp(firstConfigureForTx.blockNumber);
+
+    // Set a weight of 0.
+    const secondFundingCycleData = createFundingCycleData({ weight: ethers.BigNumber.from(0) });
+
+    // The metadata value doesn't affect the test.
+    const secondFundingCycleMetadata = ethers.BigNumber.from(234);
+
+    //fast forward to within the cycle.
+    //keep 5 seconds before the end of the cycle so make all necessary checks before the cycle ends.
+    await fastForward(firstConfigureForTx.blockNumber, firstFundingCycleData.duration.sub(5));
+
+    // Configure second funding cycle
+    const secondConfigureForTx = await jbFundingCycleStore
+      .connect(controller)
+      .configureFor(PROJECT_ID, secondFundingCycleData, secondFundingCycleMetadata, fundingCycleMustStartOnOrAfterZero);
+
+    // The timestamp the second configuration was made during.
+    const secondConfigurationTimestamp = await getTimestamp(secondConfigureForTx.blockNumber);
+
+    const expectedSecondFundingCycle = {
+      number: ethers.BigNumber.from(2), // second cycle
+      configuration: secondConfigurationTimestamp,
+      basedOn: firstConfigurationTimestamp, // based on the first cycle
+      start: firstConfigurationTimestamp.add(firstFundingCycleData.duration), // starts at the end of the first cycle
+      duration: secondFundingCycleData.duration,
+      weight: firstFundingCycleData.weight.div(1 / discountRate), // expect a weight derived from the previous cycle's values because 0 was sent. 
+      discountRate: secondFundingCycleData.discountRate,
+      ballot: secondFundingCycleData.ballot,
+      metadata: secondFundingCycleMetadata
+    };
+
+    // The `queuedOf` should contain the properties of the current cycle, with a new number, start, and weight.
+    expect(cleanFundingCycle(await jbFundingCycleStore.queuedOf(PROJECT_ID))).to.eql(expectedSecondFundingCycle);
+  });
+
+  it("Should apply a discount rate to a subsequent cycle that rolls over", async function () {
+    const { controller, mockJbDirectory, jbFundingCycleStore, addrs } = await setup();
+    await mockJbDirectory.mock.controllerOf.withArgs(PROJECT_ID).returns(controller.address);
+
+    const discountRate = 0.5; // Use a discount rate of 50%
+
+    const discountRateFidelity = 100000000; // Discount rates are stored out of this number
+
+    const fundingCycleData = createFundingCycleData({ discountRate: ethers.BigNumber.from(discountRateFidelity * discountRate) });
+
+    // The metadata value doesn't affect the test.
+    const fundingCycleMetadata = ethers.BigNumber.from(123);
```

Comment:
In all of the places where I see this, you may just want to declare some constants. FUNDING_CYCLE_METADATA[1,2,3]. Would save some repeated lines, and you could just use in line with `configureFor`.

Reply:
got it